### ISSUE TITLE
DFPManager.load(slotId) does not limit loaded slots to the slotId argument

### ISSUE THREAD

**Issue by tmcleroy** [jaanauati/react-dfp#110]
The issue is that `DFPManager.load(slotId)` loads every slot, not just the one with the given slot id as expected.

These lines seems like they may be the cause of the issue https://github.com/jaanauati/react-dfp/blob/master/js/manager.js#L160-L167

```
    if (loadAlreadyCalled === true) {
      const slot = registeredSlots[slotId];
      if (slot !== undefined) {
        availableSlots[slotId] = slot;
      }
    } else {
      availableSlots = registeredSlots;
    }
```

It seems that the available slots are only limited to the given `slotId` if `loadAlreadyCalled` is true. Is this intentional or a bug? If intentional, how can I load a single slot?

**Comment by jaanauati** [comment]
Hi @tmcleroy, I was actually not aware of this edge case, and yes, it definitively seems to be a bug.

I will push a fix soon, but i'm still a little curious on how your app works, are you declaring all your ad slots beforehand, and initializing them one by one manually?. If possible i'd suggest you to use the <DFPSlotsProvider /> as demonstrated in the following example: (work in progress) http://react-dfp.tk/examples/basic/. In this way you could declare dynamically any arbitrary number of slots and the mentioned component would initialize those AdSlots for u.

Thanks.

**Comment by tmcleroy** [comment]
Thanks for the quick response. In my app we need to avoid rendering the ad in a slot until the spot where it will be rendered is within 300px of the viewport. I have each ad slot wrapped in an Observable  component (from react-intersection-observer) whose visibility change handler will trigger the loading of the ad slot so that it's only rendered when it's in the viewport.

**Comment by jaanauati** [comment]
@tmcleroy, I've just published 0.10.0 which has a fix for this.

Additionally, load(...) now takes any number of arbitrary arguments, so you can do things like `DFPManager.load('slot-1', 'slot-3')`.

Please, give it a try and let me know if it works fine.

thanks.

**Comment by tmcleroy** [comment]
@jaanauati Awesome, everything works as expected now. Thanks again.
